### PR TITLE
[Weather/Restaurant Booking/ToDo] Align to skill template

### PR DESCRIPTION
**Pull request by ssss141414**
### Purpose
*What is the context of this pull request? Why is it being done?*

### Changes
*Are there any changes that need to be called out as significant or particularly difficult to grasp? (Include illustrative screenshots for context if applicable.)*

### Tests
*Is this covered by existing tests or new ones? If no, why not?*

### Feature Plan
*Are there any remaining steps or dependencies before this issue can be fully resolved? If so, describe and link to any relevant pull requests or issues.*

#### General
- [ ] I have commented my code, particularly in hard-to-understand areas	
- [ ] I have added or updated the appropriate tests	
- [ ] I have updated related documentation

### CODE REVIEW

File: skills/csharp/experimental/weatherskill/Dialogs/ForecastDialog.cs [hunk 1]
```diff
@@ -338,9 +328,9 @@ private string GetImageUri(string imagePath)
...
         }
 
-        private Task<DialogTurnResult> End(WaterfallStepContext stepContext, CancellationToken cancellationToken)
+        private Task<DialogTurnResult> EndAsync(WaterfallStepContext stepContext, CancellationToken cancellationToken)
         {
-            return stepContext.EndDialogAsync();
+            return stepContext.EndDialogAsync(cancellationToken: cancellationToken);
         }
 
         private class DialogIds
```
Comment (bobokids) [inline skills/csharp/experimental/weatherskill/Dialogs/ForecastDialog.cs:336]: static

**Comment by bobokids** [comment]
>         private class DialogIds

static

---
Refers to: skills/csharp/experimental/weatherskill/Dialogs/SkillDialogBase.cs:216 in 17eee6c. [](commit_id = 17eee6ce7bdb60d4927c57b7e12f0ec6abeda5fe, deletion_comment = False)

**Comment by xhr0804** [comment]
>                 .UseEndpoints(endpoints => endpoints.MapControllers());

under this line should add two lines:
```
            // Uncomment this to support HTTPS.
            // app.UseHttpsRedirection();
```

---
Refers to: skills/csharp/experimental/restaurantbookingskill/Startup.cs:148 in 50893bb. [](commit_id = 50893bbc1d9f307e7cd4bb56a449a9b5a695a4e9, deletion_comment = False)

**Comment by xhr0804** [comment]
>   },

luis field can be removed?

---
Refers to: skills/csharp/todoskill/appsettings.json:27 in 50893bb. [](commit_id = 50893bbc1d9f307e7cd4bb56a449a9b5a695a4e9, deletion_comment = False)

File: skills/csharp/todoskill/Dialogs/DeleteToDoItemDialog.cs [hunk 2]
```diff
@@ -195,81 +186,81 @@ protected async Task<DialogTurnResult> DeleteTask(WaterfallStepContext sc, Cance
                 if (canDeleteAnotherTask)
                 {
                     cardReply.InputHint = InputHints.IgnoringInput;
-                    await sc.Context.SendActivityAsync(cardReply);
+                    await sc.Context.SendActivityAsync(cardReply, cancellationToken);
                     return await sc.NextAsync();
```
Comment (xhr0804) [inline skills/csharp/todoskill/Dialogs/DeleteToDoItemDialog.cs:190]: missed "cancellationToken"

**Comment by xhr0804** [comment]
>                     return await sc.ReplaceDialogAsync(Actions.CollectTaskIndexForDelete);

missed "cancellationToken"

---
Refers to: skills/csharp/todoskill/Dialogs/DeleteToDoItemDialog.cs:365 in 50893bb. [](commit_id = 50893bbc1d9f307e7cd4bb56a449a9b5a695a4e9, deletion_comment = False)

**Comment by xhr0804** [comment]
> using System;

add Copyright announcement?

---
Refers to: skills/csharp/todoskill/Dialogs/MainDialog.cs:1 in 50893bb. [](commit_id = 50893bbc1d9f307e7cd4bb56a449a9b5a695a4e9, deletion_comment = False)

File: skills/csharp/todoskill/Dialogs/ToDoSkillDialogBase.cs [hunk 3]
```diff
@@ -217,29 +214,29 @@ protected async Task<DialogTurnResult> ClearContext(WaterfallStepContext sc, Can
                     state.TaskContentML = null;
                     state.TaskContent = null;
                     state.CollectIndexRetry = false;
-                    await DigestToDoLuisResult(sc);
+                    await DigestToDoLuisResultAsync(sc, cancellationToken);
                 }
 
                 return await sc.NextAsync();
```
Comment (xhr0804) [inline skills/csharp/todoskill/Dialogs/ToDoSkillDialogBase.cs:220]: missed "cancellationToken"

**Comment by ssss141414** [comment]
Fixed all comments, thanks for your review!

**Pull request merged**